### PR TITLE
ephemeral + improved help style

### PR DESCRIPTION
**Pull request by tanner0101**
- truncates loading bar messages (no longer spams console if too narrow)
- improved `--help` style
- push/pop ephemeral state. when you call `pushEphemeral()` all texted outputted afterward will be cleared when `popEphemeral()` is called

- [x] requires `Extendable` be conformed to `: class` cc/@LoganWright

### CODE REVIEW

File: Sources/Console/Console/Console+Ephemeral.swift [hunk 1]
```diff
@@ -0,0 +1,50 @@
+extension ConsoleProtocol {
+    public func pushEphemeral() {
+        depth += 1
+        // Swift.print("PUSH \(depth)\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n")
```
Comment (LoganWright) [inline Sources/Console/Console/Console+Ephemeral.swift]: rm

File: Sources/Console/Console/Console+Ephemeral.swift [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        levels[depth] = 0
+    }
+    
+    public func popEphemeral() {
+        let lines = levels[depth] ?? 0
+        // Swift.print("POP \(depth) = \(lines)\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n")
```
Comment (LoganWright) [inline Sources/Console/Console/Console+Ephemeral.swift]: rm

File: Sources/Console/Console/Console+Ephemeral.swift [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        get { return extend["ephemeral-levels"] as? [Int: Int] ?? [:] }
+        set { extend["ephemeral-levels"] = newValue }
+    }
+    
+    internal func didOutputLines(count: Int) {
+        guard depth > 0 else {
+            return
+        }
+        
+        // Swift.print("DID OUTPUT \(depth): +\(count)")
```
Comment (LoganWright) [inline Sources/Console/Console/Console+Ephemeral.swift]: rm

File: Sources/Console/Console/Console+Print.swift [hunk 4]
```diff
@@ -1,30 +1,79 @@
 extension ConsoleProtocol {
-    public func printUsage(executable: String, commands: [Runnable]) {
+    private func printUsage(executable: String) {
```
Comment (tanner0101) [inline Sources/Console/Console/Console+Print.swift]: `public`

File: Sources/Console/Console/Console+Print.swift [hunk 5]
```diff
@@ -1,30 +1,79 @@
...
     public func printHelp(executable: String, command: Command) {
         command.printUsage(executable: executable)
+        print("")
         printHelp(command.help)
+        print("")
         command.printSignatureHelp()
     }
 
-    public func printHelp(_ help: [String]) {
+    private func printHelp(_ help: [String]) {
```
Comment (tanner0101) [inline Sources/Console/Console/Console+Print.swift]: `public`

File: Sources/Console/Console/Console+Print.swift [hunk 6]
```diff
@@ -1,30 +1,79 @@
...
-                return command.id
-                }.joined(separator: "|"), newLine: false)
-            print(">")
-        } else {
-            print()
+        print("\(executable) ", newLine: false)
+        warning("command")
+    }
+    
+    private func printRunnable(_ commands: [Runnable]) {
```
Comment (tanner0101) [inline Sources/Console/Console/Console+Print.swift]: `public`

**Pull request merged**